src/Bridges/Constraint/map.jl

```diff
@@ -274,19 +274,39 @@ end
         bridge::AbstractBridge,
         func::MOI.AbstractFunction,
         set::MOI.AbstractSet,
+        is_available::Function,
     )
 
 Return a new constraint index `ci` and store the mapping `ci => bridge`.
+If `func isa MOI.VectorOfVariables` then `ci` is such that `is_available(ci)`,
+otherwise, `is_available` is ignored because there can only be a clash of
+indices between variable and constraint bridge mapping for `MOI.VectorOfVariables`.
 """
 function add_key_for_bridge end
 
+_ensure_available(::Map, ::Type, ::Type, ::Function) = nothing
+function _ensure_available(
+    map::Map,
+    F::Type{MOI.VectorOfVariables},
+    ::Type{S},
+    is_available::Function,
+) where {S}
+    while !is_available(MOI.ConstraintIndex{F,S}(-length(map.bridges) - 1))
```

Comment:
So this is a bit confusing. We're just going to push in a bunch of `nothing` until the index becomes available?

What if this ends up adding thousands of elements?

Reply:
It means you also have thousands of constrained vector of variables. Another solution would be to have one common vector shared between `Constraint.Map` and `Variable.Map`

Reply:
It just kind of break encapsulation so it's a tradeoff

Reply:
Fair enough. Let's wait until it's a problem.